Shell/TheoryAxioms.cpp

```diff
@@ -957,154 +957,154 @@ void TheoryAxioms::addBooleanArrayWriteAxioms(TermList arraySort)
  */
 void TheoryAxioms::apply()
 {
-  if (env.options->alasca()) {
-    addAlascaAxioms();
-    return;
-  }
   Property* prop = _prb.getProperty();
   bool modified = false;
-  bool haveIntPlus =
-    prop->hasInterpretedOperation(Theory::INT_PLUS) ||
-    prop->hasInterpretedOperation(Theory::INT_UNARY_MINUS) ||
-    prop->hasInterpretedOperation(Theory::INT_LESS) ||
-    prop->hasInterpretedOperation(Theory::INT_MULTIPLY);
-  bool haveIntMultiply =
-    prop->hasInterpretedOperation(Theory::INT_MULTIPLY);
-
-  bool haveIntDivision =
-    prop->hasInterpretedOperation(Theory::INT_QUOTIENT_E) || // let's ignore the weird _F and _T for now!
-    prop->hasInterpretedOperation(Theory::INT_REMAINDER_E) ||
-    prop->hasInterpretedOperation(Theory::INT_ABS);
-
-  bool haveIntDivides = prop->hasInterpretedOperation(Theory::INT_DIVIDES);
-
-  bool haveIntFloor = prop->hasInterpretedOperation(Theory::INT_FLOOR);
-  bool haveIntCeiling = prop->hasInterpretedOperation(Theory::INT_CEILING);
-  bool haveIntRound = prop->hasInterpretedOperation(Theory::INT_ROUND);
-  bool haveIntTruncate = prop->hasInterpretedOperation(Theory::INT_TRUNCATE);
-  bool haveIntUnaryRoundingFunction = haveIntFloor || haveIntCeiling || haveIntRound || haveIntTruncate;
-
-  if (haveIntPlus || haveIntUnaryRoundingFunction || haveIntDivision || haveIntDivides) {
-    TermList zero(theory->representConstant(IntegerConstantType(0)));
-    TermList one(theory->representConstant(IntegerConstantType(1)));
-    if(haveIntMultiply || haveIntDivision || haveIntDivides) {
-      addAdditionOrderingAndMultiplicationAxioms(Theory::INT_PLUS, Theory::INT_UNARY_MINUS, zero, one,
-						 Theory::INT_LESS, Theory::INT_MULTIPLY);
-      if(haveIntDivision){
-        addIntegerDivisionWithModuloAxioms(Theory::INT_PLUS, Theory::INT_UNARY_MINUS, Theory::INT_LESS,
-                                 Theory::INT_MULTIPLY, Theory::INT_QUOTIENT_E, Theory::INT_DIVIDES,
-                                 Theory::INT_REMAINDER_E, Theory::INT_ABS, zero,one);
-      }
-      else if(haveIntDivides){ 
-        Stack<TermList>& ns = env.signature->getDividesNvalues(); 
-        Stack<TermList>::Iterator nsit(ns);
-        while(nsit.hasNext()){
-          TermList n = nsit.next();
-          addIntegerDividesAxioms(Theory::INT_DIVIDES,Theory::INT_MULTIPLY,zero,n); 
+  if (env.options->alasca()) {
```

Comment:
Can you recall what the change was here?

What does the ALASCA branch get which it did not before? I guess it want's to do arithmetic stuff its own way, but maybe other axioms should still be added as usual (FOOL, term algebras, arrays?)

Reply:
This was the fix which reenabled all non-arithmetic theory axioms when `alasca=on`. In the end, we use default `alasca=off` but the theory axioms are still not used in the other case. We should be a bit careful though that I did not introduce any errors upon merging with master.

Reply:
The term algebra functionality should be covered by some of the new sanity checks (see `checks/term-algebra`). Otherwise, if I ask GitHub to ignore whitespaces in the diff, it seems OK -- we branch on ALASCA or old arithmetic axioms, after which come array and term algebra axioms.